raitools/raitools/_managers/counterfactual_manager.py

```diff
@@ -2,33 +2,204 @@
+    """An exception indicating that a duplicate counterfactual configuration
+    was detected.
+
+    :param exception_message: A message describing the error.
+    :type exception_message: str
+    """
+    _error_code = 'Duplicate counterfactual configuration detected.'
+
+
+class UserConfigValidationException(Exception):
+    """An exception indicating that some user configuration is not valid.
+
+    :param exception_message: A message describing the error.
+    :type exception_message: str
+    """
+    _error_code = 'Invalid config'
+
+
+class _CounterfactualConfig:
+    def __init__(self, method, continuous_features, total_CFs,
+                 desired_class='opposite', desired_range=None,
+                 permitted_range=None, features_to_vary=None):
+        self.method = method
+        self.continuous_features = continuous_features
+        self.total_CFs = total_CFs
+        self.desired_range = desired_range
+        self.desired_class = desired_class
+        self.permitted_range = permitted_range
+        self.features_to_vary = features_to_vary
+        self.is_computed = False
+        self.counterfactual_obj = None
+        self.has_computation_failed = False
+        self.failure_reason = None
+
+    def __eq__(self, other_cf_config):
+        return (
+            self.method == other_cf_config.method and
+            self.continuous_features == other_cf_config.continuous_features and
+            self.total_CFs == other_cf_config.total_CFs and
+            self.desired_range == other_cf_config.desired_range and
+            self.desired_class == other_cf_config.desired_class and
+            self.permitted_range == other_cf_config.permitted_range and
+            self.features_to_vary == other_cf_config.features_to_vary
+        )
 
 
 class CounterfactualManager(BaseManager):
-    def __init__(self):
-        pass
+    def __init__(self, model, train, test, target_column, task_type):
+        """Defines the CounterfactualManager for generating counterfactuals
+           from a model.
+
+        :param model: The model to generate counterfactuals from.
+            A model that implements sklearn.predict or sklearn.predict_proba
+            or function that accepts a 2d ndarray.
+        :type model: object
+        :param initialization_examples: A matrix of feature vector
+            examples (# examples x # features) for initializing the explainer.
+        :type initialization_examples: pandas.DataFrame
+        :param evaluation_examples: A matrix of feature vector
+            examples (# examples x # features) on which to explain the
+            model's output.
+        :type evaluation_examples: pandas.DataFrame
+        :param target_column: The name of the label column.
+        :type target_column: str
+        :param task_type: Task type is either 'classification/regression'
+        :type task_type: str
+        """
+        self._model = model
+        self._train = train
```

Comment:
Should we start using init and eval here so that we're consistent throughout all RAI code?

Reply:
Will probably keep these like above as rai analyzer code also class these as train and test. https://github.com/microsoft/responsible-ai-widgets/blob/f217a54932daf99115a2de2aee17015e26b861c0/raitools/raitools/raianalyzer/rai_analyzer.py#L43